src/cider/nrepl/print_method.clj

```diff
@@ -25,6 +25,13 @@
        (do ~@(map #(list '.write 'w %) strings))
        (#'clojure.core/print-object ~arg ~'w))))
 
+;;; Atoms
+;; Ex: #atom[{:foo :bar} 0x54274a2b]
+(def-print-method Atom c
+  "#atom["
+  (print-str @c)
```

Comment:
This should be `pr-str`, otherwise strings will print wrong (as you can see in the test).